src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2448,21 +2617,33 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
 
     addon_profiles = instance.addon_profiles or {}
     if 'kube-dashboard' in addon_args and 'kubeDashboard' not in addon_profiles:
-        addon_profiles['kubeDashboard'] = ManagedClusterAddonProfile(enabled=True)
+        addon_profiles['kubeDashboard'] = ManagedClusterAddonProfile(
+            enabled=True)
 
     os_type = 'Linux'
 
     # for each addons argument
     for addon_arg in addon_args:
-        addon = ADDONS[addon_arg]
+        if addon_arg in ADDONS:
+            addon = ADDONS[addon_arg]
+        else:
+            matches = fuzzy_match(addon_arg, list(ADDONS))
+            if len(matches) == 0:
+                matches = list(ADDONS)
+            matches = str(matches)[1:-1]
+            raise CLIError(
+                f"The addon \"{addon_arg}\" is not a recognized addon option. Possible options: {matches}")
```

Comment:
I think we need to separate the fuzzy result and all the possible options in the error message.
if fuzzy doesn't match any, we return the current error.
"The addon \"{addon_arg}\" is not a recognized addon option. Possible options: {all addons}"
if fuzzy matches any, we should return 
"The addon \"{addon_arg}\" is not a recognized addon option. Do you mean {matches}? Possible options: {all addons}"